Troubleshooting.md

```diff
@@ -0,0 +1,84 @@
+
+ ![ status indicator ](https://raw.githubusercontent.com/redhat-developer/vscode-java/master/images/statusMarker.png).
+
+### Try:
+1. If you get the error `"The JAVA_HOME environment variable points to a missing folder"` or `"Java runtime could not be located"`, please make sure the JAVA_HOME and bin path point to a valid JDK home directory. Otherwise, ignore this step.
+2. Open VS Code Command Palette (`Ctrl+Shift+P`), and run command `"Java: Update project configuration"` to force the Language Support extension to resolve the project relationship again.
+3. If the step above doesn't work, please try to clean the stale workspace cache. Close all opened VS Code windows first and delete the following cache directory:
+- Windows : %APPDATA%\Code\User\workspaceStorage\
+- MacOS : $HOME/Library/Application Support/Code/User/workspaceStorage/
+- Linux : $HOME/.config/Code/User/workspaceStorage/
+4. Try more [troubleshooting guide](https://github.com/redhat-developer/vscode-java/wiki/Troubleshooting) from the Language Support extension.
+
+## Failed to resolve classpath: xxx.
+### Reason:
+In launch mode, Java Debugger will auto resolve the classpath based on the given `"mainClass"` and `"projectName"`, if the mainClass or projectName is not correctly configured, it probably throws this kind of error.
+
+### Try:
+1. Check the mainClass is correct. The illegal class name is like `"C:\todoapp\main\src\org\microsoft\app\Main.java"`. The debugger doesn't accept the file name as mainClass. It should be `"${packageName}.${className}"` with `"."` seperated, e.g. `"org.microsoft.app.Main"`.
+2. Check the projectName is correct. The projectName is not always same as the workspace folder name. Usually the project name is read from the `"projectDescription/name"` field in the `".project"` file, or the `"artificatId"` field in the `"pom.xml"`.
+3. If the problem still exists, please use the auto generation feature provided by Java Debugger. Remove the existing launch.json file, and click F5, then Java Debugger will auto resolve the debug configurations into a new generated launch.json.
+
+## Request type "xxx" is not supported. Only "launch" and "attach" are supported.
+### Reason:
+The configurations in the launch.json are illegal.
+
+### Try:
+1. Reference the VSCode official document [launch configurations](https://code.visualstudio.com/docs/editor/debugging#_launch-configurations) about how to configure launch.json.
+2. Another alternative workaround is to remove the existing launch.json file, and click F5, then Java Debugger will auto generate the launch.json for you.
+
+## Failed to complete hot code replace: xxx.
+### Reason:
+This error indicates you are doing `Hot Code Replace`. The `Hot Code Replace` feature depends on the underlying JVM implementation. If you get this error, that indicates the new changes cannot be hot replaced by JVM.
+
+### Try:
+1. Restart your application to apply the new changes. Or ignore the message, and continue to debug.

```

Comment:
Add more doc: 

2. You can disable the hot code replace feature by change the user setting: `"java.debug.settings.enableHotCodeReplace": false`